doc/installation/system/inifile.rst

```diff
@@ -226,13 +226,17 @@ privacyIDEA Nodes
 -----------------
 
 privacyIDEA can run in a redundant setup. For statistics and monitoring purposes you
-can give these different nodes, dedicated names.
+can give these different nodes dedicated names.
 
-``PI_NODE`` is a string with the name of this very node. ``PI_NODES`` is a list of
-all available nodes in the cluster.
+``PI_NODE`` is a string with the name of this very node. At the startup of
+the application, an installation specific unique ID will be used to tie the
+node name to an installation. A distinct ID can be given to an application with
+the ``PI_UUID`` value (should conform to
+`RFC 4122 <https://datatracker.ietf.org/doc/html/rfc4122.html>`_).
+The names of all nodes in a redundant setup will be available through the database.
```

Comment:
While elaborating on PI_NODE; you totally dropped the description of PI_NODES!

Reply:
Well you see. I stepped into this trap.
So maybe we should add a sentence like

"In preveious versions you could use the key PI_NODES; which is not necessary anymore."

We might also need to add a note about decommissioning a node.
If one node is removed from a redundant setup. Looks like the admin needs to delete the entry from the database?

Also: A setup might start with "localnode" and the name of the node could be changed later. Hm, I will see if it is updated via the uuid --- most probably ;-)

Reply:
Yeah, these are the tricky parts...
1. We do not now if a node becomes/is offline, we don't have any heartbeat functionality. The only possible solution would be to discard entries based on the timestamp. However the timestamp only gets updated when privacyIDEA is (re-)started, not during every request (i would like to avoid that overhead).
2. Currently it is possible to have nodes with the same node name (not a unique requirement in the db). We could log a warning and/or attach part of the UUID to the name if the same node name is detected. If it bothers the admin, they can change it.
3. We have no "deprecation" or "migration" of `PI_NODES` (since each node fills the table themselves, i don't see any possible migration). We could check for the configuration entry during startup and log a warning.
4. In the future we could add some management functionality for the nodes which could also check for connectivity of the nodes on demand.